lgc/patch/NggPrimShader.cpp

```diff
@@ -3277,10 +3278,16 @@ void NggPrimShader::runCopyShader(Module *module) {
 
   static const unsigned CopyShaderSysValueCount = 11; // Fixed layout: 10 SGPRs, 1 VGPR
   for (unsigned i = 0; i < CopyShaderSysValueCount; ++i) {
-    if (i == CopyShaderUserSgprIdxVertexOffset)
+    if (i == 0) {
+      // Set global table (must always be the first one of user data SGPRs
+      auto userData = sysValueStart + EsGsSpecialSysValueCount;
```

Comment:
If you are indexing ver `sysValueStart`, why not pass it as `ArrayRef<Argument *>`?

Reply:
I think I should modify the parameter type from `Argument*` to `arg_iterator` in a refactoring change. `sysValueStart ` is actually `arg_iterator`, not an array of arguments.

Reply:
Or possibly some argument range to make sure you are not out of bounds.

Reply:
Good idea. I will pass the `iterator_range ` to the method call. Will have a separate PR refactoring this.